### PR TITLE
Improves local dev setup to account for system differences

### PR DESCRIPTION
**Pull request by dennisgove**
This change adds functionality to start the service and database using docker-compose thus mitigating issues that may come up if developers have differently configured systems.

### CODE REVIEW

File: lib/is-database-ready.js
Comment (jasnell) [inline lib/is-database-ready.js]: Don't call `process.exit()` from within an async function. Instead, set the `process.exitCode` here if absolutely necessary. Or, likely better, have this just return `false` since the semantic is: If there's an error the database is not ready.

File: lib/is-database-ready.js
Comment (jasnell) [inline lib/is-database-ready.js]: Just curious.. why call `process.exit()` at all? All that is needed is to set `process.exitCode = isReady ? 0 : 1` and allow the process exit normally on it's own. (sorry if being pedantic, but calling `process.exit()` in this way is just a bad practice)

Comment (dennisgove) [reply lib/is-database-ready.js]: No particular reason. I was just following the example in migrate.js, but I've seen that a PR has been submitted to change that as well. I'll reflect that change here too.

**Comment by dennisgove** [comment]
I believe all review comments/suggestions have been dealt with. It still needs 2 approvals before merge.

**Review by jasnell — APPROVED** [review]

File: .nodemonrc
Comment (mikegwhit) [inline .nodemonrc:2]: I'm unfamiliar with performance impacts of file watches.  At least, would hope we also have meaningful ignores.  Otherwise if goal is to expand file watches to outside files this is understandable.

**Review by ShaunBaker — APPROVED** [review]
Nice work - LGTM

File: Dockerfile-api
Comment (mikegwhit) [inline Dockerfile-api:4]: Uncertain but AFAIK are there cascading changes as a result of this?  WORKDIR sets the active directory, but AFAIK this may only impact the image itself?

File: docker-compose.yml
Comment (mikegwhit) [inline docker-compose.yml:25]: Just looking for clarification on what this line does?  This maps the WORKDIR to the `covid-green` volume correct?

Comment (dennisgove) [reply docker-compose.yml:25]: Yes, that's correct.

File: .nodemonrc
Comment (dennisgove) [reply .nodemonrc:2]: Previously `nodemon` was only watching files in `lib` and the `.env` file. This change causes `nodemon` to watch for changes on any files in the current directory. The performance impact during local dev didn't appear to be noticable, or at least it did restart the process when I edited `./index.js`

File: Dockerfile-api
Comment (dennisgove) [reply Dockerfile-api:4]: The only downstream issues would be things inside the container. Basically this just changes the folder the application code lives in and is run from. 

I made this change only as an attempt to keep names consistent.

Comment (dennisgove) [reply Dockerfile-api:4]: Technically, it should be noted that if anyone is running the image and mounting a file in (perhaps a `.env` file) then they would be impacted by this. But the standard way of passing env variables into a node process in a container is too set actual ENV variables, not pass a file in.

**Pull request merged**
